endpoints/api/namespacequota.py

```diff
@@ -0,0 +1,328 @@
+"""
+Manage organizations, members and OAuth applications.
+"""
+
+import logging
+
+from flask import request
+
+import features
+from auth.permissions import (
+    AdministerOrganizationPermission,
+    SuperUserPermission,
+    OrganizationMemberPermission,
+    UserReadPermission,
+)
+from data import model
+from data.model import config
+from endpoints.api import (
+    resource,
+    nickname,
+    ApiResource,
+    validate_json_request,
+    request_error,
+    require_user_admin,
+    show_if,
+)
+from endpoints.exception import InvalidToken, Unauthorized
+
+logger = logging.getLogger(__name__)
+
+
+def quota_view(orgname: str, quota, quota_limit_types):
+    return {
+        "orgname": orgname,
+        "limit_bytes": quota.limit_bytes if quota else None,
+        "quota_limit_types": quota_limit_types,
+    }
+
+
+def quota_limit_view(orgname: str, quota_limit):
+    return {
+        "percent_of_limit": quota_limit["percent_of_limit"],
+        "limit_type": {
+            "name": quota_limit["name"],
+            "quota_limit_id": quota_limit["id"],
+            "quota_type_id": quota_limit["type_id"],
+        },
+    }
+
+
+def namespace_size_view(orgname: str, repo):
+    return {
+        "orgname": orgname,
+        "repository_name": repo.name,
+        "repository_size": repo.repositorysize.size_bytes,
+        "repository_id": repo.id,
+    }
+
+
+@resource("/v1/namespacequota/<namespace>/quota")
+@show_if(features.QUOTA_MANAGEMENT)
+class OrganizationQuota(ApiResource):
+
+    schemas = {
+        "NewOrgQuota": {
+            "type": "object",
+            "description": "Description of a new organization quota",
+            "required": ["limit_bytes"],
+            "properties": {
+                "limit_bytes": {
+                    "type": "integer",
+                    "description": "Number of bytes the organization is allowed",
+                },
+            },
+        },
+    }
+
+    @nickname("getNamespaceQuota")
+    def get(self, namespace):
+        orgperm = OrganizationMemberPermission(namespace)
+        userperm = UserReadPermission(namespace)
+
+        if not orgperm.can() and not userperm.can():
+            raise Unauthorized()
```

Comment:
We have predefined scope decorator you could use to handle permissions.
For example `require_repo_read`, `require_repo_write`, `require_repo_admin` from `endpoints/api/__init__`

Reply:
Ah. I guess it's because both user and org quotas share the same endpoint.
Wondering if these should be split? Especially if we were to expand the feature set of quotas on orgs in the future. For example, quotas per member, etc. In which case it would be better for it to orgs and users to have their own quota subresource.

Reply:
I'd argue that it would be better to do it now. That way we don't introduce major api changes again in the future.
Also, currently, most of our users and orgs apis separate, so it would be consistent to manage quotas the same way.